pkg/gadgets/trace/exec/types/types.go

```diff
@@ -15,19 +15,40 @@
 package types
 
 import (
+	"fmt"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/columns"
 	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
 )
 
 type Event struct {
 	eventtypes.Event
 
-	Pid       uint32   `json:"pid,omitempty"`
-	Ppid      uint32   `json:"ppid,omitempty"`
-	UID       uint32   `json:"uid,omitempty"`
-	MountNsID uint64   `json:"mountnsid,omitempty"`
-	Retval    int      `json:"ret,omitempty"`
-	Comm      string   `json:"pcomm,omitempty"`
-	Args      []string `json:"args,omitempty"`
+	Pid       uint32   `json:"pid,omitempty" column:"pid,width:7"`
+	Ppid      uint32   `json:"ppid,omitempty" column:"ppid,width:7"`
+	UID       uint32   `json:"uid,omitempty" column:"uid,width:7"`
+	MountNsID uint64   `json:"mountnsid,omitempty" column:"mntns,width:12,hide"`
+	Retval    int      `json:"ret,omitempty" column:"ret,width:4"`
+	Comm      string   `json:"pcomm,omitempty" column:"comm,width:16"`
+	Args      []string `json:"args,omitempty" column:"args,width:24"`
+}
+
+// execColumns is defined as global variable so that the callers panic if they
+// are importing this package with invalid column tags. However, force callers
+// to use MustGetColumns to include additional configuration to the columns.
+var execColumns = columns.MustCreateColumns[Event]()
```

Comment:
Is it a really good idea to put this on an external variable? What about users that don't need it at all, it seems like a waste of memory to me.

Reply:
Yeah, it is not worth it for those who are not interested in using the column format.